clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaProducerFactory.java

```diff
@@ -59,4 +61,15 @@
      */
     Future<Void> closeProducer(String producerName);
 
+    /**
+     * Sets Kafka metrics support with which producers created by this factory will be registered.
+     * <p>
+     * Factory implementations may choose to not support this feature.
+     *
+     * @param metricsSupport The metrics support.
+     */
+    default void setMetricsSupport(final KafkaClientMetricsSupport metricsSupport) {
```

Comment:
IMHO this shouldn't be a default implementation as it doesn't do anything.

Reply:
Changed this into a normal interface method.